utils/ixdtf/src/parsers/grammar.rs

```diff
@@ -6,170 +6,170 @@
+pub(crate) fn is_duration_designator(ch: char) -> bool {
     ch == 'P' || ch == 'p'
 }
 
 /// Checks if char is a `YearDesignator`.
 #[inline]
 #[cfg(feature = "duration")]
-pub(crate) const fn is_year_designator(ch: char) -> bool {
+pub(crate) fn is_year_designator(ch: char) -> bool {
     ch == 'Y' || ch == 'y'
 }
 
 /// Checks if char is a `MonthsDesignator`.
 #[inline]
 #[cfg(feature = "duration")]
-pub(crate) const fn is_month_designator(ch: char) -> bool {
+pub(crate) fn is_month_designator(ch: char) -> bool {
     ch == 'M' || ch == 'm'
 }
 
 /// Checks if char is a `WeekDesignator`.
 #[inline]
 #[cfg(feature = "duration")]
-pub(crate) const fn is_week_designator(ch: char) -> bool {
+pub(crate) fn is_week_designator(ch: char) -> bool {
     ch == 'W' || ch == 'w'
 }
 
 /// Checks if char is a `DayDesignator`.
 #[inline]
 #[cfg(feature = "duration")]
-pub(crate) const fn is_day_designator(ch: char) -> bool {
+pub(crate) fn is_day_designator(ch: char) -> bool {
     ch == 'D' || ch == 'd'
 }
 
 /// checks if char is a `DayDesignator`.
 #[inline]
 #[cfg(feature = "duration")]
-pub(crate) const fn is_hour_designator(ch: char) -> bool {
+pub(crate) fn is_hour_designator(ch: char) -> bool {
     ch == 'H' || ch == 'h'
 }
 
 /// Checks if char is a `MinuteDesignator`.
 #[inline]
 #[cfg(feature = "duration")]
-pub(crate) const fn is_minute_designator(ch: char) -> bool {
+pub(crate) fn is_minute_designator(ch: char) -> bool {
     is_month_designator(ch)
 }
 
 /// checks if char is a `SecondDesignator`.
 #[inline]
 #[cfg(feature = "duration")]
-pub(crate) const fn is_second_designator(ch: char) -> bool {
+pub(crate) fn is_second_designator(ch: char) -> bool {
     ch == 'S' || ch == 's'
 }
 
 /// Checks if char is a `DecimalSeparator`.
 #[inline]
-pub(crate) const fn is_decimal_separator(ch: char) -> bool {
+pub(crate) fn is_decimal_separator(ch: char) -> bool {
     ch == '.' || ch == ','
 }
 
 /// Checks if char is an `AnnotationOpen`.
 #[inline]
-pub(crate) const fn is_annotation_open(ch: char) -> bool {
+pub(crate) fn is_annotation_open(ch: char) -> bool {
     ch == '['
 }
 
 /// Checks if char is an `AnnotationClose`.
 #[inline]
-pub(crate) const fn is_annotation_close(ch: char) -> bool {
+pub(crate) fn is_annotation_close(ch: char) -> bool {
     ch == ']'
 }
 
 /// Checks if char is an `CriticalFlag`.
 #[inline]
-pub(crate) const fn is_critical_flag(ch: char) -> bool {
+pub(crate) fn is_critical_flag(ch: char) -> bool {
     ch == '!'
 }
 
 /// Checks if char is the `AnnotationKeyValueSeparator`.
 #[inline]
-pub(crate) const fn is_annotation_key_value_separator(ch: char) -> bool {
+pub(crate) fn is_annotation_key_value_separator(ch: char) -> bool {
     ch == '='
 }
 
 /// Checks if char is a hyphen. Hyphens are used as a Date separator
 /// and as a `AttributeValueComponent` separator.
 #[inline]
-pub(crate) const fn is_hyphen(ch: char) -> bool {
+pub(crate) fn is_hyphen(ch: char) -> bool {
```

Comment:
Observation: you can make these `const` again (but there's not really a need to)

Reply:
Yeah, I noticed those could be flipped back to `const`, but I figured there wasn't much of a gain either way (might do it though on an upcoming commit to lower the diff)